### PR TITLE
Feature/config base webpack

### PR DESCRIPTION
**Pull request by HuyAms**
- feat: add webpack files
- feat: add clean-webpack-plugin to remove dist folder before each build
- feat: add file-loader to load fonts
- feat: add url-loader and image-webpack-loader to handle images
- feat: add dist folder to gitignore
- feat: add publicPath to output
- refactor: put webpack files to webpack folder

### CODE REVIEW

File: webpack/webpack.dev.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+module.exports = merge(common, {
+  mode: 'development',
+
+  output: {
+    filename: '[name].js',
+    chunkFilename: '[name].chunk.js',
+    publicPath: '/',
+  },
+
```
Comment (vinhlee95) [inline webpack/webpack.dev.js]: `devtool` and `devServer` could be for debugging. Check this out:
https://webpack.js.org/configuration/devtool/

My previous project's configs: 

```javascript
devtool: 'eval-source-map',
	devServer: {
		stats: {
			colors: true,
			hash: false,
			version: false,
			timings: true,
			assets: false,
			chunks: false,
			modules: false,
			reasons: false,
			children: false,
			source: true,
			errors: true,
			errorDetails: false,
			warnings: false,
			publicPath: false
		},
		// server default port
		port: 3001,
		historyApiFallback: true
	},
```

File: webpack/webpack.prod.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+module.exports = merge(common, {
+  mode: 'production',
+
+  output: {
+    filename: '[name].[chunkhash].js',
+    chunkFilename: '[name].[chunkhash].chunk.js',
+    publicPath: '/',
+  },
+
```
Comment (vinhlee95) [inline webpack/webpack.prod.js]: Should we add some plugins for optimization in this stage? 
https://webpack.js.org/guides/production/

File: .gitignore [hunk 3]
```diff
@@ -7,6 +7,7 @@ scripts/flow/*/.flowconfig
 _SpecRunner.html
 __benchmarks__
 build/
+dist/
 remote-repo/
 coverage/
 .module-cache
```
[the remark this answers is not in the document]
Comment (HuyAms) [reply .gitignore]: No we should not ignore package-lock. It saves all package versions

File: webpack/webpack.dev.js (hunk 1 above)
Comment (HuyAms) [reply webpack/webpack.dev.js]: I will add it when config webpack for development

File: webpack/webpack.prod.js (hunk 2 above)
Comment (HuyAms) [reply webpack/webpack.prod.js]: It's for the webpack-production story

**Pull request merged**